transmit-lib/src/cmr/transmit/generic_documents.clj

```diff
@@ -18,38 +18,27 @@
    [ring.util.codec :as codec]))
 
 (defn- concept-ingest-url
-  "Generate a URL to the metadata db for posting to the generic APIs"
-  ([conn provider-id]
-   {:pre [conn provider-id]}
-   (println "in concept ingest url")
-   (format "%s/generics/%s/%s" (conn/root-url conn) (codec/url-encode provider-id)(str (java.util.UUID/randomUUID))))
-  ([conn provider-id concept-id]
-   {:pre [conn concept-id]}
-   (format "%s/generics/%s/%s/%s"
-           (conn/root-url conn)
-           (codec/url-encode provider-id)
-           (codec/url-encode concept-id)
-           (str (java.util.UUID/randomUUID)))))
-
-(defn- concept-ingest-url2
   "Generate a URL to the metadata db for posting to the generic APIs, but allow
    for a list of parameters to be passed in which will then be applied again to
    this function. This is to allow for the original calling function to specify
    both provider-id and concept-id while using the CRUD macros."
   ([conn param-list]
    {:pre [conn param-list]}
-   (apply concept-ingest-url2 (into [conn] param-list)))
-  
-  ([conn provider-id concept-id]
-   {:pre [conn provider-id concept-id]}
-   (concept-ingest-url conn provider-id concept-id)))
+   (apply concept-ingest-url (into [conn] param-list)))
+
+  ([conn provider-id native-id]
+   {:pre [conn provider-id native-id]}
+   (format "%s/generics/%s/%s"
+           (conn/root-url conn)
+           (codec/url-encode provider-id)
+           (codec/url-encode native-id))))
 
 ;; These are "standard" macros for creating CRUD functions. get, update, and
 ;; destroy all require 2 parameters and must use a url resolving function that
 ;; does an apply with an array of paramaters to resolve them. The calling
 ;; function must pass in the URL parameters as an array. Create can use the
 ;; single param function.
 (http-helper/defcreator2 create-generic :metadata-db concept-ingest-url {:use-system-token? true :raw? true})
-(http-helper/defgetter read-generic :metadata-db concept-ingest-url2 {:use-system-token? true :raw? true})
-(http-helper/defupdater update-generic :metadata-db concept-ingest-url2 {:use-system-token? true})
-(http-helper/defdestroyer delete-generic :metadata-db concept-ingest-url2 {:use-system-token? true})
+(http-helper/defgetter read-generic :metadata-db concept-ingest-url {:use-system-token? true :raw? true})
+(http-helper/defupdater update-generic :metadata-db concept-ingest-url {:use-system-token? true})
+(http-helper/defdestroyer delete-generic :metadata-db concept-ingest-url {:use-system-token? true})
```

Comment:
Good to see you removed url2.  As much as I like this clean code. It's hard to read. and the description doesn't really help much.  Could you give an example of what function it creates and how it's being called?

Reply:
- concept-ingest-url is responsible for returning a URL for a service and it will take an array of parameters for that URL
- http-helper/def* macros all create a function which makes an HTTP call to an internal CMR service. There are some assumptions like getter will use GET and destroyer will use DELETE. The built function will pass in common headers a service like metadata-db would expect. Many of the functions above this block of code might be able to be converted over to using these macros. I found the macro being used in only a handful of places and decided that since I was creating new interfaces, I would try to use just these macros. An example usage is in `cmr.ingest.api.generic-documents/read-generic-document`.